From 187f33729504ecec76604b5fbea56117f6bf391b Mon Sep 17 00:00:00 2001
From: Sujeet Baranwal <sbaranwal@marvell.com>
Date: Tue, 11 Jun 2019 11:03:49 -0700
Subject: [PATCH 329/386] firmware: arm_scmi: enable polling mode

octeontx2's AP is communicating with SCP using polling mode so the
necessary flags are enabled.

Change-Id: Ide581ad0a7a576032d93d8cf848d7809c593a17e
Signed-off-by: Sujeet Baranwal <sbaranwal@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/10748
Reviewed-by: Chandrakala Chavva <cchavva@marvell.com>
Reviewed-by: Sujeet Kumar Baranwal <Sujeet.Baranwal@cavium.com>
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/firmware/arm_scmi/driver.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/firmware/arm_scmi/driver.c b/drivers/firmware/arm_scmi/driver.c
index 8f952f2f1a29..886ac740cdad 100644
--- a/drivers/firmware/arm_scmi/driver.c
+++ b/drivers/firmware/arm_scmi/driver.c
@@ -393,6 +393,10 @@ int scmi_do_xfer(const struct scmi_handle *handle, struct scmi_xfer *xfer)
 	if (unlikely(!cinfo))
 		return -EINVAL;
 
+#ifdef CONFIG_MVL_MHU
+	/* pollng mode */
+	xfer->hdr.poll_completion = true;
+#endif
 	ret = mbox_send_message(cinfo->chan, xfer);
 	if (ret < 0) {
 		dev_dbg(dev, "mbox send fail %d\n", ret);
@@ -477,6 +481,10 @@ int scmi_xfer_get_init(const struct scmi_handle *handle, u8 msg_id, u8 prot_id,
 	xfer->hdr.protocol_id = prot_id;
 	xfer->hdr.poll_completion = false;
 
+#ifdef CONFIG_MVL_MHU
+	/* Enable polling mode */
+	xfer->hdr.poll_completion = true;
+#endif
 	*p = xfer;
 
 	return 0;
-- 
2.17.1

